docs/lesson-plans/week-4.md

```diff
@@ -0,0 +1,623 @@
+the path. Another image with a more complex URL will be shown below which dissects more components.
+
+### Routing in Vert.x
+
+Routing in Vertx is done with `Router` and `Route` objects provided by the Vert.x dependency. In the api directory, we 
+have an `ApiMain` class, which handles creating a `Router` Vert.x object, setting up a lot of the routing settings we'll 
+need in the future, creating a home `Route`, and setting the port to start on.
+
+The home `Route` or root route is the path `/`. Using what we learned earlier, if we were hosting a server locally on 
+port :8081, that would be accessible by the URL `http://localhost:8081/`. We can also see that in the `Route` object for
+home produced by the `Router` in `ApiMain.startApi()`, a route handler was set using the `handler()` method and by 
+passing in a [Method Reference](https://docs.oracle.com/javase/tutorial/java/javaOO/methodreferences.html), which is a 
+type of lambda or anonymous function. Down below, you can see a `handleHome()` method that takes in a `RoutingContext`
+from Vert.x, which is an object containing all of the request info we need. We use that to construct a response by 
+getting a new response object from the `RoutingContext`, setting the status code, putting header information, and 
+setting CORS info. Finally, the interaction is ended with the `HttpServerResponse.end()` method.
+
+Since that's a lot of stuff to set when finishing up a response, we've provided some `end()` methods for you in the 
+`com.codeforcommunity.rest.IRouter` interface, which will add a lot of that for you! A lot of that was done manually in 
+the `handleHome()` method to show what it looks like.
+
+#### Static Routing
+
+Let's go through the exercise of creating a new route. We'll need to start out by creating a new `Route` object from the
+`Router`. First what you should do is create a new one right after the home route's handler is set. We'll do that using 
+a register route method. You'll need to pass in the `Router` object that was created earlier in the method. 
+Let's create a route `/hi`.
+
+```java 
+home.handler(this::homeHandler);
+
+// Create a new Route object from the Router, and do it for a GET request.
+this.registerGetHiRoute(router);
+
+...
+// Down below in a new method.
+private void registerGetHiRoute(Router router) {
+  // Set the route up to respond to GET method requests.
+  Route hiRoute = router.get("/hi");
+  ...
+```
+
+Next we'll set the handler for that request method and finish up the `registerGetHiRoute` method.
+
+```java 
+private void registerGetHiRoute(Router router) {
+  // Set the route up to respond to GET method requests.
+  Route hiRoute = router.get("/hi");
+  // Note: The hiRouteHandler method doesn't exist yet...
+  hiRoute.handler(this::hiRouteHandler);
+}
+```
+
+Finally, we'll return some data for the route in the `hiRouteHandler` method. We'll make sure to do it using our 
```

Comment:
Finally, we'll define the `hiRouteHandler` method to return some data for the route.